File: AvoidAGrabCutEasy/ProcOutline/ProgressEventArgs.cs
using System;

namespace GetAlphaMatte
{
    public class ProgressEventArgs : EventArgs
    {
        private int _prgInterval = 1;
        public int ImgWidthHeight
        {
            get
            {
                return m_ImgWidthHeight;
            }
            set
            {
                m_ImgWidthHeight = value;
            }
        }
        private int m_ImgWidthHeight;
        public double CurrentProgress
        {
            get
            {
                return m_CurrentProgress;
            }
            set
            {
                m_CurrentProgress = value;
            }
        }
        private double m_CurrentProgress;
        public int PrgInterval
        {
            get
            {
                return _prgInterval;
            }
            set
            {
                _prgInterval = Math.Max(value, 1);
            }
        }

        public ProgressEventArgs(int ImageWidthHeight, int StartValue)
        {
            ImgWidthHeight = ImageWidthHeight;
            CurrentProgress = StartValue;
        }

        public ProgressEventArgs(int ImageWidthHeight, double StartValue, int Interval)
        {
            ImgWidthHeight = ImageWidthHeight;
            CurrentProgress = StartValue;
            PrgInterval = Interval;
        }
    }
}

File: FloatPointPxBitmap/ProgressEventArgs.vb
Option Strict On

Public Class ProgressEventArgs
    Inherits EventArgs

    Private _prgInterval As Integer = 1
    Public Property ImgWidthHeight() As Integer
        Get
            Return m_ImgWidthHeight
        End Get
        Set(value As Integer)
            m_ImgWidthHeight = value
        End Set
    End Property
    Private m_ImgWidthHeight As Integer
    Public Property CurrentProgress() As Double
        Get
            Return m_CurrentProgress
        End Get
        Set(value As Double)
            m_CurrentProgress = value
        End Set
    End Property
    Private m_CurrentProgress As Double
    Public Property PrgInterval() As Integer
        Get
            Return _prgInterval
        End Get
        Set(value As Integer)
            _prgInterval = Math.Max(value, 1)
        End Set
    End Property

    Public Sub New(ImageWidthHeight As Integer, StartValue As Integer)
        ImgWidthHeight = ImageWidthHeight
        CurrentProgress = StartValue
    End Sub

    Public Sub New(ImageWidthHeight As Integer, StartValue As Double, Interval As Integer)
        ImgWidthHeight = ImageWidthHeight
        CurrentProgress = StartValue
        PrgInterval = Interval
    End Sub
End Class
